satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -0,0 +1,662 @@
+:class:`FiduceoMviriBase` for a keyword argument description. Global
+netCDF attributes are available in the ``raw_metadata`` attribute of
+each loaded dataset.
+
+
+Image Orientation
+-----------------
+The images are stored in MVIRI scanning direction, that means South is up and
+East is right. This can be changed as follows:
+
+.. code-block:: python
+
+    scn.load(['VIS'], upper_right_corner='NE')
+
+
+Geolocation
+-----------
+In addition to the image data, FIDUCEO also provides so called *static FCDRs*
+containing latitude and longitude coordinates. In order to simplify their
+usage, the FIDUCEO MVIRI readers do not make use of these static files, but
+instead provide an area definition that can be used to compute longitude and
+latitude coordinates on demand.
+
+.. code-block:: python
+
+    area = scn['VIS'].attrs['area']
+    lons, lats = area.get_lonlats()
+
+Those were compared to the static FCDR and they agree very well, however there
+are small differences. The mean difference is < 1E3 degrees for all channels
+and projection longitudes.
+
+
+Huge VIS Reflectances
+---------------------
+You might encounter huge VIS reflectances (10^8 percent and greater) in
+situations where both radiance and solar zenith angle are small. The reader
+certainly needs some improvement in this regard. Maybe the corresponding
+uncertainties can be used to filter these cases before calculating reflectances.
+
+
+Angles
+------
+The FIDUCEO MVIRI FCDR provides satellite and solar angles on a coarse tiepoint
+grid. By default these datasets will be interpolated to the higher VIS
+resolution. This can be changed as follows:
+
+.. code-block:: python
+
+    scn.load(['solar_zenith_angle'], resolution=4500)
+
+If you need the angles in both resolutions, use data queries:
+
+.. code-block:: python
+
+    from satpy import DataQuery
+
+    query_vis = DataQuery(
+        name='solar_zenith_angle',
+        resolution=2250
+    )
+    query_ir = DataQuery(
+        name='solar_zenith_angle',
+        resolution=4500
+    )
+    scn.load([query_vis, query_ir])
+
+    # Use the query objects to access the datasets as follows
+    sza_vis = scn[query_vis]
+
+
+References
+----------
+    - `[Handbook]`_ MFG User Handbook
+    - `[PUG]`_ FIDUCEO MVIRI FCDR Product User Guide
+
+.. _[Handbook]: http://www.eumetsat.int/\
+website/wcm/idc/idcplg?IdcService=GET_FILE&dDocName=PDF_TD06_MARF&\
+RevisionSelectionMethod=LatestReleased&Rendition=Web
+.. _[PUG]: http://doi.org/10.15770/EUM_SEC_CLM_0009
+"""
+
+import abc
+import functools
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.dataset.dataid import DataQuery
+from satpy.readers._geos_area import (ang2fac, get_area_definition,
+                                      get_area_extent)
+from satpy.readers.file_handlers import BaseFileHandler
+
+EQUATOR_RADIUS = 6378140.0
+POLE_RADIUS = 6356755.0
+ALTITUDE = 42164000.0 - EQUATOR_RADIUS
```

Comment:
The `FULL` data contains these values in the file:
- `header2_EarthEquatorialRadius`
- `header2_EarthPolarRadius`
- `header3_SatEarcthCentreDistance` (typo is in the data, not an accident here)

Would it be feasible to update the reader to take advantage of these values if they're available? Or is that too tricky as they're only in the FULL and not EASY files...

Reply:
I like the idea, but there's a slight difference between the values from the header and these constants (which I copied from the MFG Handbook). Another question for EUM :slightly_smiling_face: